pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -364,6 +364,57 @@ func fixLibvirtCrcNetworkAvailable() error {
 	return nil
 }
 
+func removeLibvirtCrcNetwork() error {
+	logging.Debug("Removing libvirt 'crc' network")
+	_, _, err := crcos.RunWithDefaultLocale("virsh", "--connect", "qemu:///system", "net-info", libvirt.DefaultNetwork)
+	if err == nil {
+		cmd := exec.Command("virsh", "--connect", "qemu:///system", "net-destroy", libvirt.DefaultNetwork)
+		err = cmd.Run()
+		buf := new(bytes.Buffer)
+		cmd.Stderr = buf
+		if err != nil {
+			logging.Debugf("%v : %s", err, buf.String())
+			return fmt.Errorf("Failed to destroy libvirt 'crc' network")
+		}
+
+		cmd = exec.Command("virsh", "--connect", "qemu:///system", "net-undefine", libvirt.DefaultNetwork)
+		err = cmd.Run()
+		buf = new(bytes.Buffer)
+		cmd.Stderr = buf
+		if err != nil {
+			logging.Debugf("%v : %s", err, buf.String())
+			return fmt.Errorf("Failed to undefine libvirt 'crc' network")
+		}
+	}
+	logging.Debug("libvirt 'crc' network removed")
+	return nil
+}
+
+func removeCrcVM() error {
+	logging.Debug("Removing 'crc' VM")
+	_, _, err := crcos.RunWithDefaultLocale("virsh", "--connect", "qemu:///system", "dumpxml", constants.DefaultName)
+	if err == nil {
+		cmd := exec.Command("virsh", "--connect", "qemu:///system", "destroy", constants.DefaultName)
+		err := cmd.Run()
+		buf := new(bytes.Buffer)
+		cmd.Stderr = buf
+		if err != nil {
+			logging.Debugf("%v : %s", err, buf.String())
+			return fmt.Errorf("Failed to destroy 'crc' VM")
+		}
+		cmd = exec.Command("virsh", "--connect", "qemu:///system", "undefine", constants.DefaultName)
+		err = cmd.Run()
+		buf = new(bytes.Buffer)
+		cmd.Stderr = buf
+		if err != nil {
+			logging.Debugf("%v : %s", err, buf.String())
+			return fmt.Errorf("Failed to undefine 'crc' VM")
+		}
+	}
+	logging.Debug("'crc' vm is removed")
+	return nil
+}
```

Comment:
I think we also want to remove `~/.crc/machines/crc` after destroying the VM. And I wonder if we should also removed the cached bundle when it's embedded in our binary? As crc setup will extract it.

Reply:
@cfergeau yes, we are going to remove `~/.crc/machines/crc` but I am not sure if we should also remove the cache bits because then we are again adding extraction time to user side if he is on the same version.